unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -146,11 +145,23 @@ public async UniTask<FriendRequest> RejectFriendshipAsync(string friendRequestId
         public void RemoveFriend(string friendId) =>
             apiBridge.RemoveFriend(friendId);
 
-        public void GetFriends(int limit, int skip) =>
-            apiBridge.GetFriends(limit, skip);
+        public async UniTask<string[]> GetFriendsAsync(int limit, int skip, CancellationToken ct)
+        {
+            var payload = await apiBridge.GetFriendsAsync(limit, skip, ct);
+            TotalFriendCount = payload.totalFriends;
+            AddFriends(payload.friends);
 
-        public void GetFriends(string usernameOrId, int limit) =>
-            apiBridge.GetFriends(usernameOrId, limit);
+            return payload.friends;
+        }
+
+        public async UniTask<string[]> GetFriendsAsync(string usernameOrId, int limit, CancellationToken ct)
+        {
+            var payload= await apiBridge.GetFriendsAsync(usernameOrId, limit, ct);
```

Comment:
it is not needed to switch to main thread after this operation?

Reply:
It is not needed but the IDE gives you a warning due to this.

Reply:
Yes! I forgot it! Done, thanks!